packages/amplify_datastore/example/lib/main.dart

```diff
@@ -293,7 +297,8 @@ class _MyAppState extends State<MyApp> {
   deletePost(String id) async {
     try {
       _selectedPostForNewComment = null;
-      await Amplify.DataStore.delete(Post(id: id, title: null));
+      await Amplify.DataStore.delete(
+          Post(id: id, title: "", rating: 0, created: TemporalDateTime.now()));
```

Comment:
Sure that's a great point.  And leads us to other considerations as well.  

The delete really just needs the id field set to know which model to delete.  But because of null safety now requiring fields to be set, we have to set a bunch of junk fields to just delete correctly ... This isn't good user experience.  

We probably need to have a meeting for this actually where we consider if we should have a "justId" constructor for this deletion case or if we can update the delete method to just take an id. 

@haverchuck @Ashish-Nanda what are your thoughts on this?

In the meantime, as this is a relic of the past as you said, I think in the interests of time we should look to resolve this in another PR.  Perhaps a hashmap of id->model instances.

Reply:
@cshfang @fjnoyp The docs from [amplify-android ](https://docs.amplify.aws/lib/datastore/data-access/q/platform/android#delete) assume the use of a full instance to delete, while[ amplify-ios](https://docs.amplify.aws/lib/datastore/data-access/q/platform/ios#delete) seems to describe the ability to delete with an instance, plus a 'justId'-type method. We could potentially do the ios version as well, but the lack of method overloading would be a challenge here.  Either we have a second api method (we tend to have a high bar for introducing these), or we would use optional parameters in a very hacky way - correct me if you see any alternatives.  

In any case, to @fjnoyp's point about "a bunch of junk fields", we should probably change our docs to show the query-then-delete pattern that native recommends if we go forward with this as-is.

Reply:
Sounds good.  I spoke with Ashish yesterday about it and we agreed we'll go with this as is for now.  

We'll make a note of it for now and address it later down the road.